projects/admin/src/app/classes/typeahead/mef-typeahead.ts

```diff
@@ -111,36 +191,75 @@ export class MefTypeahead implements ITypeahead {
     if (null == options?.filters?.selected) {
       throw Error('Missing filters definition');
     }
-
     const searchCategory = options.filters.selected;
-    // TODO :: Why do I need to add an ending slash to use proxy-config and don't be redirected (308) receiving a CORS errors
+
+    // Build API call to get remote suggestions.
     const url = `${this._apiSearchEntryPoint}/${searchCategory}/${searchTerm}/`;
     const headers = new HttpHeaders({ 'Content-Type': 'application/json' });
-
-    return this._http.get(url, {headers}).pipe(
+    const remoteSuggestions$ = this._http.get(url, {headers}).pipe(
       map((results: any) => results?.hits?.total >= 0 ? results.hits.hits : []),
-      map((hits: Array<any>) => {
-        const suggestions = [];
-        hits.map((hit: any) => {
-          for (const source of this._sources()) {
-            if (hit.metadata[source]) {
-              suggestions.push(this._getNameRef(hit.metadata, source));
-            }
-          }
-        });
-        return suggestions;
-      }),
+      map((hits: Array<any>) => this._buildRemoteSuggestions(hits)),
+      catchError(e => {
+        switch (e.status) {
+          case 400: return of([]);
+          default: throw e;
+        }
+      })
+    );
+    // Build API call to get local entity suggestions.
+    const localSuggestions$ = this._recordService.getRecords('local_entities', searchTerm, 1, 10, undefined, {type: searchCategory}).pipe(
+      map((result: Record) => this._recordService.totalHits(result.hits.total) === 0 ? [] : result.hits.hits),
+      map((hits: Array<any>) => this._buildLocalSuggestions(hits)),
       catchError(e => {
         switch (e.status) {
-          case 400:
-            return of([]);
-          default:
-            throw e;
+          case 400: return of([]);
+          default: throw e;
         }
       })
     );
+    // Return the concatenation of both API request.
+    return forkJoin([remoteSuggestions$, localSuggestions$]).pipe(
+      map(([remoteHits, localHits]) => [...remoteHits, ...localHits])
+    );
```

Comment:
general remark: why not to have one api like `api/entities/suggestions` to retrieve all formatted suggestions ?
😜